fly/commands/internal/executehelpers/uploads.go

```diff
@@ -19,7 +19,18 @@ func Upload(bar *mpb.Bar, team concourse.Team, path string, includeIgnored bool,
 	archiveStream, archiveWriter := io.Pipe()
 
 	go func() {
-		archiveWriter.CloseWithError(tarfs.Compress(zstd.NewWriter(archiveWriter), path, files...))
+		zstdWriter, err := zstd.NewWriter(archiveWriter)
+		if err != nil {
+			return
```

Comment:
should we bubble this err up?

Reply:
I think if we close the pipe's writer with the error here, we might end up already propagating it from what I got - when doing `archiveWriter.CloseWithError(err)`, that gets stored, and then subsequently a `read` on the other side would notice that it's closed, and then bring the error.

(didn't verify that myself)